src/primitives/zerocoin.cpp

```diff
@@ -3,6 +3,88 @@
 // file COPYING or http://www.opensource.org/licenses/mit-license.php.
 
 #include "primitives/zerocoin.h"
+#include "init.h"
+#include "util.h"
+#include "wallet.h"
+
+void CZerocoinMint::ZerocoinCrypterSetup() {
+    CKey key;
+    CPubKey vchPubKey;
+
+    if(pwalletMain->IsLocked()) {
+        LogPrintf("%s : failed to encrypt zerocoin mint since the wallet is locked\n", __func__);
+        return;
+    }
+
+    assert(CReserveKey(pwalletMain).GetReservedKey(vchPubKey)); //should never fail
+
+    if (!pwalletMain->GetKey(vchPubKey.GetID(), key)) {
+        LogPrintf("%s : failed to encrypt zerocoin mint. Could not find key %s\n", __func__,
+                  vchPubKey.GetHash().ToString().c_str());
+        return;
+    }
+
+    //inialize IV for 32 bytes
+    vector<unsigned char> chIV(crypter.KEY_SIZE);
```

Comment:
You should follow our current naming conventions. A vector will start with `v` and camel case after. A vector of unsigned chars should start with `vch`. Descriptive names are also a good idea.

Reply:
This function appears to use no member variables of CZeromint at all (plus it uses a global pointer) so I wonder why it is not a free function or part of the CWallet class. Personally I would have added this, Encrypt and Decrypt as separate functions so there would be separate plain CZeromint objects and encrypted objects.  I don't see the benefit of encrypting an object if the un-encrypted member variables are still accessible. However, with separate objects you can dispose of the un-encrypted ones in favor of the encrypted (once they've been encrypted) or vice-versa. This may just be a lack of understanding of the bigger picture here but it just doesn't seem like a good separation of concerns. Hope this is clear. So essentially a separate layer or separate encrypted objects that is all done outside of this class